### ISSUE TITLE
[Docs] incorrect pre-requirements for Fedora

### ISSUE THREAD

**Issue by Jan200101** [Cog-Creators/Red-DiscordBot#2558]
the [docs](https://red-discordbot.readthedocs.io/en/v3-develop/install_linux_mac.html#centos-7-fedora-and-rhel) state that the following commands need to be run on Fedora and possibly RHEL

I'm going to adress every command on its own

> yum -y groupinstall development

the development group group is called `Development Tools` but its questionable if all those tools are really needed for this bot, if they are needed please correct me in a comment

> yum -y install https://centos7.iuscommunity.org/ius-release.rpm

the repo is designed for CentOS 7 and adding it to Fedora introduces a bunch of new packages which are not really needed and just slow down dnf (yum on Fedora is a dnf wrapper)

> sudo yum install zlib-devel bzip2 bzip2-devel readline-devel sqlite sqlite-devel \
openssl-devel xz xz-devel libffi-devel git2u java-1.8.0-openjdk

everything is fine here except git2u
Fedora already includes a recent version of git (git version 2.20.1 as of writing) so there is no reason at all to install the ius repo

I'd be willing to make a seperate pre-requirements list for Fedora needed

**Comment by mikeshardmind** [comment]
I'll rewrite the CentOS/RHEL section and remove Fedora from it. You're correct about not needing everything in the development tools group. 

This section was written with CentOS in mind for the most part, and knowing the instructions can translate to RHEL, but they should not be broadly applied to Fedora.

**Comment by mikeshardmind** [comment]
Looking more into this, we should be splitting our docs by major distribution again. The fact that we require python3.7 now fragments instructions enough that we should be ensuring each distribution has separate docs which full cover what end users need.

**Comment by Tobotimus** [comment]
In #2920 I've added a section for Fedora Linux, and, given that version 29 and above has Python 3.7 in official repositories, it's very straightforward.

For now, I decided to keep development tools, since it adds a failsafe when one of our dependencies which include C extensions doesn't have a wheel available. For example, this would probably happen if someone installed Fedora on an ARM machine (not as unusual as you might think - RPi 3 and above users looking to take advantage of their 64-bit armv8 CPU would need to look away from Raspbian). From what I gather, `@development-tools` is the equivalent of `build-essential` and linux headers on Debian - we install `build-essential` with all of the debian-based distributions as well.

In case you were wondering, these dependencies have C extensions: 
- aiohttp (provides wheels on win/mac/linux for x86 and x86-64)
- python-Levenshtein (we provide wheels on win/mac/linux for x86, x86-64 and ARM)
- PyYAML (only provides wheels on Windows)
- uvloop (only gets installed on non-Windows systems,  provides wheels on mac/linux for x86 and x86-64)
- websockets (provides wheels on win/mac/linux for x86 and x86-64)
- yarl (provides wheels on win/linux for x86 and x86-64)
- pymongo (mongo extra only, provides wheels on win/mac/linux for x86 and x86-64)
- asyncpg (future, postgres extra only, provides wheels on win/mac/linux for x86 and x86-64)

On Raspbian, some of these packages may be available on [piwheels](https://www.piwheels.org), but it's not guaranteed.
